apps/remix-ide/src/app/tabs/hardhat-provider.js

```diff
@@ -58,7 +55,8 @@ const sendAsyncInternal = (provider, data, resolve, reject) => {
       resolve(message)
     })
   } else {
-    resolve({ jsonrpc: '2.0', result: [], id: data.id })
+    const result = data.method === 'net_listening' ? 'canceled' : []
```

Comment:
actually maybe a better way would have been to `reject` if the provider is not set..

Reply:
`reject` will also show a predefined error on toaster